src/main/java/at/hannibal2/skyhanni/features/fishing/FishingAPI.kt

```diff
@@ -52,11 +61,18 @@ object FishingAPI {
     fun onItemInHandChange(event: ItemInHandChangeEvent) {
         // TODO correct rod type per island water/lava
         holdingRod = event.newItem.isFishingRod()
+        holdingLavaRod = event.newItem.isLavaRod()
+        holdingWaterRod = event.newItem.isWaterRod()
     }
 
-    fun isLavaRod() = InventoryUtils.getItemInHand()?.getLore()?.any { it.contains("Lava Rod") } ?: false
+    @SubscribeEvent
+    fun onRepoReload(event: RepositoryReloadEvent) {
+        val data = event.getConstant<ItemsJson>("Items")
+        lavaRods = data.lava_fishing_rods ?: error("§clava_fishing_rods is missing from repo.")
+        waterRods = data.water_fishing_rods ?: error("§cwater_fishing_rods is missing from repo.")
```

Comment:
We dont need these errors as it will already throw an error and say the repo status isnt ideal if we try to set these lists to a null element

Reply:
But without those error messages, we don't know what part exactly of the JSON file is wrong

Reply:
It still produces an error, which will be presented to the user. Also these feel out of place as we don't have many other places where we error if one part of repo is missing. And if we do i'd like to remove those as they feel out of place. Getting an error that says something in ItemsJson is not correct should be all we need to debug and I feel the error would even say which part of the repo doesnt exist

Reply:
the error doesnt know anything about the json itself. but the error will show the stack trace, pointing to this line. so its fine and your arguments about consistency are valid.